_data-prepper/log-analytics.md

```diff
@@ -0,0 +1,149 @@
+
+There are a lot of complex Grok features that will not be discussed here, so please read the documentation if you are interested.
+
+# OpenSearch sink
+
+There is a generic sink that writes the data to OpenSearch as the destination. The [OpenSearch sink](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/README.md) has configuration options related to an OpenSearch cluster like endpoint, SSL/Username, index name, index template, index state management, etc.
+
+# Pipeline configuration
+
+## Example pipeline.yaml with SSL and basic authentication enabled
+
+Example `pipeline.yaml` with SSL and basic authentication enabled for the `http-source`:
+
+```yaml
+log-pipeline:
+  source:
+    http:
+      ssl_certificate_file: "/full/path/to/certfile.crt"
+      ssl_key_file: "/full/path/to/keyfile.key"
+      authentication:
+        http_basic:
+          username: "myuser"
+          password: "mys3cret"
+  processor:
+    - grok:
+        match:
+          # This will match logs with a "log" key against the COMMONAPACHELOG pattern (ex: { "log": "actual apache log..." } )
+          # You should change this to match what your logs look like. See the grok documenation to get started.
+          log: [ "%{COMMONAPACHELOG}" ]
+  sink:
+    - opensearch:
+        hosts: [ "https://localhost:9200" ]
+        # Change to your credentials
+        username: "admin"
+        password: "admin"
+        # Add a certificate file if you are accessing an OpenSearch cluster with a self-signed certificate  
+        #cert: /path/to/cert
+        # If you are connecting to an Amazon OpenSearch Service domain without
+        # Fine-Grained Access Control, enable these settings. Comment out the
+        # username and password above.
+        #aws_sigv4: true
+        #aws_region: us-east-1
+        # Since we are grok matching for apache logs, it makes sense to send them to an OpenSearch index named apache_logs.
+        # You should change this to correspond with how your OpenSearch indices are set up.
+        index: apache_logs
+```
+
+This pipeline configuration is an example of Apache log ingestion. Don't forget that you can easily configure the Grok processor for your own custom logs. You will need to modify the configuration above for your OpenSearch cluster.
+
+The main changes you need to make are:
+
+* `hosts` - Set to your hosts
+* `index` - Change this to the OpenSearch index you want to send logs to
+* `username`- Provide the OpenSearch username
+* `password` - Provide your OpenSearch password
+* `aws_sigv4` - If you use Amazon OpenSearch Service with AWS signing, set this to true. It will sign requests with the default AWS credentials provider.
+* `aws_region` - If you use Amazon OpenSearch Service with AWS signing, set this value to your region.
```

Comment:
Maybe clarify this:

```
[...] set this value to the region hosting your domain.
```

Or something like that. I'm not sure if "domain" is necessarily the right word, but it's important to clarify that the region someone lives in or access AWS console from may not be the same region where the cluster lives.